pagopa/io-app
# chore: [IOBP-622] Update `PaymentCard` components

## Short description
This PR updates the `PaymentCard` and `PaymentCardSmall` components, removing references to Bancomat and CoBadge payment methods.
It also updates the UI for expired cards in the new wallet section.

## List of changes proposed in this pull request
- Added required locale keys for expired tag
- Updated pagoPA API definitions
- Updated cards Design System section
- Added isExpired prop to the `PaymentCard[...]` components: this allows to configure their expire state based on the context in which they are used
- Fixed an issue with `isPaymentMethodExpired` which was causing cards to not be displayed with "expired" state
- Added tests fo `isPaymentMethodExpired` utility function

## How to test
With the `io-dev-api-server`, enable the New wallet section FF from the Profile screen. You should be able to see the expired card in your wallet. 

## Preview
https://github.com/pagopa/io-app/assets/6160324/16a5f9c5-5b36-4829-83f8-a5294ba7b126
